python/paddle/fluid/tests/unittests/mkldnn/test_activation_mkldnn_op.py

```diff
@@ -79,46 +79,70 @@ def setUp(self):
-class TestMKLDNNGeluBf16Dim2(TestActivation):
+#Use it as a base class for BF16 activation tests, just override necessary functions
+class TestMKLDNNSigmoidBF16Op(TestActivation):
+    @OpTestTool.skip_if_not_cpu_bf16()
+    def config(self):
+        self.op_type = "sigmoid"
+        self.op_func = lambda x: (1 / (1 + np.exp(-x)))
+        self.op_grad_func = lambda dout, x: (dout * self.op_func(x)) * (1 - self.op_func(x))
+
+    def set_attrs(self):
+        self.attrs = {"use_mkldnn": True}
+
+    def init_data(self):
+        self.x = np.random.uniform(-1, 1, [2, 4, 3, 5]).astype(np.float32)
+
     def setUp(self):
-        self.op_type = "gelu"
         self.dtype = np.uint16
+        self.init_data()
+        self.config()
+        self.out = self.op_func(self.x)
 
-        x = np.random.uniform(-1, 1, [11, 17]).astype(np.float32)
-        out = convert_float_to_uint16(gelu(x, False))
+        self.inputs = {'X': convert_float_to_uint16(self.x)}
+        self.outputs = {'Out': self.out}
+        self.set_attrs()
 
-        self.inputs = {'X': convert_float_to_uint16(x)}
-        self.outputs = {'Out': out}
-        self.attrs = {"use_mkldnn": True}
+    def calculate_grads(self):
+        self.dx = self.op_grad_func(self.out, self.x)
 
     def test_check_output(self):
         self.check_output_with_place(core.CPUPlace())
 
     def test_check_grad(self):
-        pass
+        self.calculate_grads()
+        self.check_grad_with_place(
+            core.CPUPlace(), ["X"],
+            "Out",
+            user_defined_grads=[self.dx],
+            user_defined_grad_outputs=[convert_float_to_uint16(self.out)])
 
 
-@unittest.skipIf(not core.supports_bfloat16(),
-                 "place does not support BF16 evaluation")
-class TestMKLDNNGeluBf16Dim2Approx(TestActivation):
-    def setUp(self):
+class TestMKLDNNGeluErfBF16Op(TestMKLDNNSigmoidBF16Op):
+    def config(self):
         self.op_type = "gelu"
-        self.dtype = np.uint16
+        self.op_func = lambda x: gelu(x, False)
+        self.op_grad_func = lambda dout, x: (dout * (0.5 + 0.5 * erf(x / np.sqrt(2)) + (x / np.sqrt(2 * np.pi) * np.exp(-0.5 * np.power(x, 2)))))
 
-        x = np.random.uniform(-1, 1, [11, 17]).astype(np.float32)
-        out = convert_float_to_uint16(gelu(x, True))
 
-        self.inputs = {'X': convert_float_to_uint16(x)}
-        self.outputs = {'Out': out}
+class TestMKLDNNGeluErfDim2BF16Op(TestMKLDNNGeluErfBF16Op):
+    def init_data(self):
+        self.x = np.random.uniform(-1, 1, [11, 17]).astype(np.float32)
+
+
+class TestMKLDNNGeluTanhBF16Op(TestMKLDNNSigmoidBF16Op):
+    def config(self):
+        self.op_type = "gelu"
+        self.op_func = lambda x: gelu(x, True)
+        self.op_grad_func = lambda dout, x: (dout * 0.5 * (1 + np.tanh(np.sqrt(2 / np.pi) * (x + 0.044715 * np.power(x, 3)))) * (1 + np.sqrt(2 / np.pi) * (x + 0.134145 * np.power(x, 3)) * (1 - np.tanh(np.sqrt(2 / np.pi) * (x + 0.044715 * np.power(x, 3))))))
```

Comment:
Please refactor this long lambda to the local function.

Reply:
I have changed all these "self.op_func" and "self.op_grad_func" occurences to local functions, because I was assigning anonymous functions to a variable with name and after some research about good practices I know that it shouldn't be done like that